OfficeDev/office-ui-fabric-react
# GroupedList: add aria-expanded and aria-controls

#### Pull request checklist

- [X] Addresses an existing issue: Fixes #5751

#### Description of changes

These changes add `aria-expanded` and `aria-controls` to the expand/collapse button in the GroupHeader of GroupedList. The `aria-expanded` details whether the region the expand/collapse button controls is expanded or collapsed. The `aria-controls` specifies which region the expand/collapse button controls.

#### Justification

Taking into account the extensive discussion we had on a previous PR for this bug, and after re-reading the W3C standards site, I feel that these changes are the right approach. On the `aria-expanded` [W3C page](https://www.w3.org/WAI/GL/wiki/Using_the_WAI-ARIA_aria-expanded_state_to_mark_expandable_and_collapsible_regions), it states the following as an example of when to use the attribute:

> Example 1: Using a button to collapse and expand a region
A button hides or exposes a paragraph of text on the page. The state of the paragraph is represented by the attribute aria-expanded on the button. 

The `aria-controls` attribute seems like a necessary addition given that the list is not a direct child of the header. The W3C standards site states:

> If the element with the aria-expanded attribute controls the expansion of another grouping container that is not 'owned by' the element, the author SHOULD reference the container by using the aria-controls attribute.

#### Focus areas to test

1. Go to the GroupedList Basic Example page. 
2. Inspect the expand/collapse button. 
3. Verify that it has an `aria-expanded` attribute and that is set correctly.
4. Verify that when the group is expanded, the `aria-controls` references the id of an element that exists.
5. Collapse the group and verify that the `aria-controls` attribute disappears along with the element that was collapsed.

These changes also updated the snapshots for all of the DetailsList tests containing a `groups` prop.
